### DISCUSSION TITLE
Async log example

### DISCUSSION THREAD

**Discussion by gumbarros** [dotnet/extensions#4105] [Q&A]
Making logging async would be very disruptive as it would mean **every** code path that calls logging would have to be async. It also means that the caller is likely *blocked* (even if via `await`) waiting for the log message to be sent.

The recommendation here is to use background worker. This is even what the console logger does (even though it's all in-process). Any logger that does heavy I/O should consider adopting this pattern:

1. The logger is synchronous and just adds messages to an internal buffer
1. A background `IHostedService` runs and collects messages from this buffer, flushing them to the I/O sink (remote service, console, file, etc.)

_Originally posted by @analogrelay in https://github.com/dotnet/extensions/issues/2141#issuecomment-580419085_

Hello @analogrelay! Can you provide an example? I don't know how to setup a buffer at my ILogger implementation.

**Comment by gumbarros** [comment]
Do you think this implementation is fine?

FileLogger:
https://github.com/JJConsulting/JJMasterData/blob/sun/src/Commons/Logging/File/FileLogger.cs

FileLoggerBackgroundService:
https://github.com/JJConsulting/JJMasterData/blob/sun/src/Commons/Logging/File/FileLoggerBackgroundService.cs

LoggerBackgroundService:
https://github.com/JJConsulting/JJMasterData/blob/sun/src/Commons/Logging/LoggerBackgroundService.cs

**Comment by RussKie** [comment]
@gumbarros the issue you're referring it to was opened against the old code that has been essentially removed. As of May 2023 this code base is completely different. I'm moving this to a discussion, but you may have better luck finding an answer to your questions in https://github.com/dotnet/runtime or https://github.com/dotnet/aspnetcore repos.

**Comment by gumbarros** [comment]
Moved discussion to https://github.com/dotnet/runtime/discussions/87949
